realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -907,66 +959,61 @@ void BaseRealSenseNode::publishFrame(
         height = timage.get_height();
         bpp = timage.get_bytes_per_pixel();
     }
-    auto& image = images[stream];
 
-    if (image.size() != cv::Size(width, height) || image.depth() != _image_formats[bpp])
+    if (image_publishers.find(stream) != image_publishers.end())
     {
-        image.create(height, width, _image_formats[bpp]);
-    }
-    image.data = (uint8_t*)f.get_data();
+        auto &image_publisher = image_publishers.at(stream);
+        cv::Mat image_cv_matrix;
+        if (_rgbd_publisher && 0 != _rgbd_publisher->get_subscription_count())
+        {
+            image_cv_matrix = getCVMatImage(f, images, width, height, bpp, stream);
+        }
+        if (0 != image_publisher->get_subscription_count())
```

Comment:
Missing some comments here.
Can we separate `rgbd` to a dedicated function?

`image_publisher` is depth image publisher?

Id we asked for rgbd we allocate the first, if we ask only depth we allocate the second?

Reply:
image_publisher is used for depth and for rgb publishers
image is used to store all matrix of depth and rgb
that what was before.
I tried to reorder the code with minimal changes, creating small functions, to reuse this in different places, specially for the RGBD.
We can sync on this over teams.